File: spoj/ucode2.0/3.py
n,a,b = list(map(int,input().split()))
s = input()
ans = 0
prev = "x"
for ch in s:
    if ch == "x":
        prev = "x"
        continue
    if prev == "x":
        if a>0 and a>b:
            ans+=1
            a-=1
            prev = "a"
        elif b>0:
            b-=1
            ans+=1
            prev = "b"
    elif prev == "a":
        if b>0:
            b-=1
            ans+=1
            prev = "b"
    elif prev == "b":
        if a>0:
            a-=1
            ans+=1
            prev = "a"
print(ans)

File: spoj/codeforces/633_2/3.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){

    int t;
    cin>>t;
    while(t--){
        int n;cin>>n;
        vector<int>arr(n);
        for(int i=0;i<n;i++){
            cin>>arr[i]; 
        }
        int ans = 0;
        int diff = 0;
        int prev = arr[0];
        for(int i=1;i<n;i++){
            diff = max(diff,prev-arr[i]);
            prev = max(prev,arr[i]);
        }
        if(diff == 0)cout<<0<<endl;
        else
        cout<<ceil(log2(diff+1))<<endl;
    }
}
